### PR TITLE
RuboCop: Fix Style/GuardClause.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/extend/ARGV.rb [hunk 1]
```diff
@@ -88,11 +88,9 @@ def kegs
             Formulary.from_rack(rack)
           end
 
-          if (prefix = f.installed_prefix).directory?
-            Keg.new(prefix)
-          else
-            raise MultipleVersionsInstalledError, rack.basename
-          end
+          raise MultipleVersionsInstalledError, rack.basename unless (prefix = f.installed_prefix).directory?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/ARGV.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an `unless .. end`.

File: Library/Homebrew/extend/ENV/shared.rb [hunk 2]
```diff
@@ -328,9 +327,7 @@ def fetch_compiler(value, source)
   end
 
   def check_for_compiler_universal_support
-    if homebrew_cc =~ GNU_GCC_REGEXP
-      raise "Non-Apple GCC can't build universal binaries"
-    end
+    raise "Non-Apple GCC can't build universal binaries" if homebrew_cc =~ GNU_GCC_REGEXP
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/ENV/shared.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/extend/string.rb [hunk 3]
```diff
@@ -59,9 +59,8 @@ def gsub!(before, after, audit_result = true)
   # Looks for Makefile style variable defintions and replaces the
   # value with "new_value", or removes the definition entirely.
   def change_make_var!(flag, new_value)
-    unless gsub!(/^#{Regexp.escape(flag)}[ \t]*=[ \t]*(.*)$/, "#{flag}=#{new_value}", false)
-      errors << "expected to change #{flag.inspect} to #{new_value.inspect}"
-    end
+    return if gsub!(/^#{Regexp.escape(flag)}[ \t]*=[ \t]*(.*)$/, "#{flag}=#{new_value}", false)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/string.rb]: Ideally split this into a few variables but no worries if you'd rather not.

File: Library/Homebrew/keg.rb [hunk 4]
```diff
@@ -423,18 +422,17 @@ def resolve_any_conflicts(dst, mode)
...
-      end
-
-      dst.unlink unless mode.dry_run
-      keg.link_dir(src, mode) { :mkpath }
-      return true
+    return unless stat.directory?
+    begin
+      keg = Keg.for(src)
+    rescue NotAKegError
+      puts "Won't resolve conflicts for symlink #{dst} as it doesn't resolve into the Cellar" if ARGV.verbose?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/keg.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/requirement.rb [hunk 5]
```diff
@@ -83,12 +83,10 @@ def modify_build_environment
...
-    if @satisfied_result.is_a?(Pathname)
-      parent = @satisfied_result.parent
-      unless ENV["PATH"].split(File::PATH_SEPARATOR).include?(parent.to_s)
-        ENV.append_path("PATH", parent)
-      end
-    end
+    return unless @satisfied_result.is_a?(Pathname)
+    parent = @satisfied_result.parent
+
+    ENV.append_path("PATH", parent) unless ENV["PATH"].split(File::PATH_SEPARATOR).include?(parent.to_s)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/requirement.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/tap.rb [hunk 6]
```diff
@@ -130,9 +130,7 @@ def git_last_commit_date
   # The issues URL of this {Tap}.
   # e.g. `https://github.com/user/homebrew-repo/issues`
   def issues_url
-    if official? || !custom_remote?
-      "https://github.com/#{user}/homebrew-#{repo}/issues"
-    end
+    "https://github.com/#{user}/homebrew-#{repo}/issues" if official? || !custom_remote?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/test/test_cmd_testbot.rb [hunk 7]
```diff
@@ -45,9 +45,7 @@ def run
   end
 
   def teardown
-    unless passed?
-      raise "INFO: Previous test failed with ENV['TRAVIS'] = #{ENV["TRAVIS"].inspect}"
-    end
+    raise "INFO: Previous test failed with ENV['TRAVIS'] = #{ENV["TRAVIS"].inspect}" unless passed?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/test_cmd_testbot.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/test/test_integration_cmds.rb [hunk 8]
```diff
@@ -39,9 +39,7 @@ def teardown
   end
 
   def needs_test_cmd_taps
-    unless ENV["HOMEBREW_TEST_OFFICIAL_CMD_TAPS"]
-      skip "HOMEBREW_TEST_OFFICIAL_CMD_TAPS is not set"
-    end
+    skip "HOMEBREW_TEST_OFFICIAL_CMD_TAPS is not set" unless ENV["HOMEBREW_TEST_OFFICIAL_CMD_TAPS"]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/test_integration_cmds.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/test/testing_env.rb [hunk 9]
```diff
@@ -57,9 +57,7 @@ def after_teardown
       super
       files_after_test = []
       Find.find(TEST_TMPDIR) { |f| files_after_test << f.sub(TEST_TMPDIR, "") }
-      if @__files_before_test != files_after_test
-        @@log.puts location, diff(@__files_before_test, files_after_test)
-      end
+      @@log.puts location, diff(@__files_before_test, files_after_test) if @__files_before_test != files_after_test
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/testing_env.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/version.rb [hunk 10]
```diff
@@ -192,11 +192,8 @@ def self.create(val)
   end
 
   def initialize(val)
-    if val.respond_to?(:to_str)
-      @version = val.to_str
-    else
-      raise TypeError, "Version value must be a string; got a #{val.class} (#{val})"
-    end
+    raise TypeError, "Version value must be a string; got a #{val.class} (#{val})" unless val.respond_to?(:to_str)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/version.rb]: Better to try and keep this below 80 chars by e.g. splitting this into an`if .. end`.

File: Library/Homebrew/extend/string.rb (hunk 3 above)
Comment (reitermarkus) [inline Library/Homebrew/extend/string.rb]: I'm leaving this alone for now.

**Comment by reitermarkus** [comment]
@MikeMcQuaid, should be good to merge now.

**Pull request merged**